tempesta_fw/ss_skb.c

```diff
@@ -691,8 +691,10 @@ __skb_fragment(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	d_size = skb_headlen(skb);
 	offset = pspt - (char *)skb->data;
 
-	if ((offset >= 0) && (offset < d_size)) {
+	if ((offset >= 0) && (offset <= d_size)) {
 		int t_size = d_size - offset;
+		if (unlikely(!t_size))
```

Comment:
Technically this is the same as point to first byte of first fragment, so why don't you consider the fast path at line 709?